cram_pr2/cram_pr2_shopping_demo/src/plans.lisp

```diff
@@ -0,0 +1,173 @@
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+(defun move-object (?object)
+    (let ((?grasping-arm :right)
+          (?grasp-pose *pose-grasping*)
+          (?grasp-pose-left *pose-grasping-left*)
+          (?grasp-pose-right *pose-grasping-right*)
+          (?pose-near-table *pose-near-table*)
+          (?table *table*)
+          (map->obj NIL)
+          (map->base NIL)
+          (trans NIL))
+
+      (exe:perform 
+       (desig:a motion (type moving-torso)
+                (joint-angle 0)))
+      
+      (exe:perform
+       (desig:an action (type going)
+                 (target (desig:a location (pose ?grasp-pose)))))
+      
+
+      (exe:perform
+       (desig:an action
+                 (type positioning-arm)
+                 (left-configuration park)
+                 (right-configuration park)))
+      
+
+      ;; creating the new Transform from the base_footprint infront of the shelf and the object
+      (setf map->base (cl-transforms:make-transform
+                       (cl-transforms:make-3d-vector -1.5 -0.3 0)
+                       (cl-transforms:axis-angle->quaternion
+                        (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+      
+      (setf map->obj (second (fourth (desig:desig-prop-value ?object :pose))))
+      (setf trans (cl-transforms:transform* (cl-transforms:transform-inv map->base) map->obj))
+
+      (setf ?newpose (cl-transforms-stamped:make-pose-stamped
+                      "base_footprint"
+                      0.0
+                      (cl-transforms:translation trans)
+                      (cl-transforms:orientation
+                       (second (first (desig:desig-prop-value ?object :pose))))))
+      
+      (setf ?newTransform (cl-transforms-stamped:make-transform-stamped
+                          "base_footprint"
+                          (desig:desig-prop-value ?object :name)
+                          0.0
+                          (cl-transforms:translation trans)
+                          (cl-transforms:rotation
+                           (second (second (desig:desig-prop-value ?object :pose))))))
+
+      ;; constructing the new Object designator with the new pose and transformation
+      (setf ?object (desig:copy-designator ?object
+                                           :new-description
+                                           `((:type ,(desig:desig-prop-value ?object :type))
+                                             (:name ,(desig:desig-prop-value ?object :name))
+                                             (:pose ((:pose ,?newpose)
+                                                     (:transform ,?newTransform)
+                                                     (:pose-in-map ,(second (third (desig:desig-prop-value ?object :pose))))
+                                                     (:transform-in-map ,(second (fourth (desig:desig-prop-value ?object :pose)))))))))
```

Comment:
Please try to stick to 80 characters per line or only a little bit over that.
Too long lines don't fit into the emacs screen with two tabs and are very annoying.
They also don't fit in the GitHub visualization screen.